clang/test/CodeGenCXX/mangle-class-nttp.cpp

```diff
@@ -27,12 +27,6 @@ template void f<B{nullptr}>();
 // CHECK: define weak_odr void @_Z1fIXtl1BLPKi32EEEEvv(
 // MSABI: define {{.*}} @"??$f@$2UB@@PEBH0CA@H0A@@@@YAXXZ"
 template void f<B{fold((int*)32)}>();
-#ifndef _WIN32
```

Comment:
Should we be retaining this test and getting a different mangling?

Reply:
Not sure I understand, with this patch, `template void f<B{fold(reinterpret_cast<int*>(0))}>();` is equivalent to the `template void f<B{nullptr}>();` from line 25, so this test causes a second explicit template instantiation.